src/management/api/analytics/logs/logging-configuration.controller.ts

```diff
@@ -33,43 +38,118 @@ class ApiLoggingConfigurationController {
+    this.initLogging(this.enabled);
     this.maxDuration = Constants.org.settings.logging.maxDurationMillis;
 
     this.$scope.loggingModes = [
       {
-        name: 'None',
-        value: 'NONE'
+        title: 'Client only',
+        id: 'CLIENT',
+        icon: 'communication:shield-user',
+        description: 'to log content between the client and the gateway'
       }, {
-        name: 'Client only',
-        value: 'CLIENT'
-      }, {
-        name: 'Proxy only',
-        value: 'PROXY'
-      }, {
-        name: 'Client and proxy',
-        value: 'CLIENT_PROXY'
-      }];
+        title: 'Proxy only',
+        id: 'PROXY',
+        icon: 'communication:shield-thunder',
+        description: 'to log content between the gateway and the backend'
+      },
+      {
+        title: 'Client & proxy',
+        id: 'CLIENT_PROXY',
+        icon: 'home:earth',
+        description: 'to log content for the client and the gateway'
+      }
+    ];
+
+    this.$scope.contentModes = [
+      {
+        title: 'Headers & payloads',
+        id: 'HEADERS_PAYLOADS',
+        icon: 'finance:selected-file',
+        description: 'to log headers and payloads'
+      },
+      {
+        title: 'Headers only',
+        id: 'HEADERS',
+        icon: 'finance:file',
+        description: 'to log headers without payloads'
+      },
+      {
+        title: 'Payloads only',
+        id: 'PAYLOADS',
+        icon: 'communication:clipboard-list',
+        description: 'to log payloads without headers'
+      },
+    ];
+
+    this.$scope.scopeModes = [
+      {
+        title: 'Request & Response',
+        id: 'REQUEST_RESPONSE',
+        icon: 'navigation:exchange',
+        description: 'to log request and response'
+      },
+      {
+        title: 'Request only',
+        id: 'REQUEST',
+        icon: 'navigation:right-2',
+        description: 'to log request content only'
+      },
+      {
+        title: 'Response only',
+        id: 'RESPONSE',
+        icon: 'navigation:left-2',
+        description: 'to log response content only'
+      },
+    ];
 
     this.$scope.$on('apiChangeSuccess', (event, args) => {
       this.api = args.api;
+      this.enabled = this.api.proxy.logging.mode !== 'NONE';
     });
   }
 
+  initLogging(enabled) {
+    if (this.api.proxy.logging == null) {
+      this.api.proxy.logging = {};
+    }
+    if (enabled) {
+      const tmp = {...this.defaultLogging, ...this.initialApi.proxy.logging};
+      const {mode, scope, content} = tmp;
+      this.api.proxy.logging.mode = mode !== 'NONE' ? mode : this.defaultLogging.mode;
+      this.api.proxy.logging.scope = scope !== 'NONE' ? scope : this.defaultLogging.scope;
+      this.api.proxy.logging.content = content !== 'NONE' ? content : this.defaultLogging.content;
+    } else {
+      this.api.proxy.logging.mode = 'NONE';
+      this.api.proxy.logging.scope = 'NONE';
+      this.api.proxy.logging.content = 'NONE';
+    }
+  }
+
+  switchMode({detail}) {
```

Comment:
Can we add a type to `detail` as it is not very clear to understand what is it, a boolean or an object 🤔 ?

```suggestion
  switchMode({detail}: {detail: boolean}) {
```